controllers/machine_reconcile_scope.go

```diff
@@ -0,0 +1,1076 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/patch"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/go-logr/logr"
+	rufiov1 "github.com/tinkerbell/rufio/api/v1alpha1"
+	tinkv1 "github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
+
+	infrastructurev1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/api/v1beta1"
+	"github.com/tinkerbell/cluster-api-provider-tinkerbell/internal/templates"
+)
+
+const (
+	providerIDPlaceholder = "PROVIDER_ID"
+	inUse                 = "in_use"
+	provisioned           = "provisioned"
+)
+
+var (
+	// ErrMachineVersionEmpty is the error returned when Version is not set on the parent Machine.
+	ErrMachineVersionEmpty = fmt.Errorf("machine version is empty")
+
+	// ErrConfigurationNil is the error returned when TinkerbellMachineReconciler or TinkerbellClusterReconciler is nil.
+	ErrConfigurationNil = fmt.Errorf("configuration is nil")
+
+	// ErrMissingClient is the error returned when TinkerbellMachineReconciler or TinkerbellClusterReconciler do
+	// not have a Client configured.
+	ErrMissingClient = fmt.Errorf("client is nil")
+
+	// ErrMissingBootstrapDataSecretValueKey is the error returned when the Secret referenced for bootstrap data
+	// is missing the value key.
+	ErrMissingBootstrapDataSecretValueKey = fmt.Errorf("retrieving bootstrap data: secret value key is missing")
+
+	// ErrBootstrapUserDataEmpty is the error returned when the referenced bootstrap data is empty.
+	ErrBootstrapUserDataEmpty = fmt.Errorf("received bootstrap user data is empty")
+
+	// ErrHardwareMissingDiskConfiguration is returned when the referenced hardware is missing
+	// disk configuration.
+	ErrHardwareMissingDiskConfiguration = fmt.Errorf("disk configuration is required")
+
+	// errWorkflowFailed is the error returned when the workflow fails.
+	errWorkflowFailed = fmt.Errorf("workflow failed")
+)
+
+type machineReconcileScope struct {
+	//*baseMachineReconcileContext
+	log               logr.Logger
+	ctx               context.Context
+	tinkerbellMachine *infrastructurev1.TinkerbellMachine
+	patchHelper       *patch.Helper
+	client            client.Client
+
+	machine              *clusterv1.Machine
```

Comment:
These 3 fields were moved from the base object to ensure the existing functions could work without alteration.